r4/src/main/java/org/opencds/cqf/r4/providers/MeasureOperationsProvider.java

```diff
@@ -222,32 +222,78 @@ public MeasureReport evaluateMeasure(@IdParam IdType theId,
+    public Parameters careGapsReport(
+        @OperationParam(name = "periodStart") List <String> periodStart,
+        @OperationParam(name = "periodEnd") List <String> periodEnd,
+        @OperationParam(name = "subject") List <String> subject,
+        @OperationParam(name = "topic") String topic,
+        @OperationParam(name = "practitioner") String practitioner,
+        @OperationParam(name = "measureId") List <String> measureId,
+        @OperationParam(name = "measureIdentifier") List <String> measureIdentifier,
+        @OperationParam(name = "measureUrl") List <CanonicalType > measureUrl,
+        @OperationParam(name = "status") List <String> status,
+        @OperationParam(name = "organization") String organization,
+        @OperationParam(name = "program") String program) {
         //TODO: topic should allow many and be a union of them
         //TODO: "The Server needs to make sure that practitioner is authorized to get the gaps in care report for and know what measures the practitioner are eligible or qualified."
         Parameters returnParams = new Parameters();
-        
-        if (careGapParameterValidation(periodStart, periodEnd, subject, topic, practitioner, measureId, measureIdentifier, measureUrl, status, organization, program)) {
-            List<Measure> measures = resolveMeasures(measureId, measureIdentifier, measureUrl);
-            if (subject.startsWith("Patient/")){
-                resolvePatientGapBundleForMeasures(periodStart, periodEnd, subject, topic, status, returnParams, measures, "return", organization);
-            } else if (subject.startsWith("Group/")) {
-                returnParams.setId((status==null?"all-gaps": status) + "-" + subject.replace("/","_") + "-report");
-                (getPatientListFromGroup(subject))
-                    .forEach(groupSubject -> resolvePatientGapBundleForMeasures(periodStart, periodEnd, subject, topic, status, returnParams, measures, "return", organization));
-            }
-            else if (Strings.isNullOrEmpty(practitioner)) {
+
+        // Setting periodStart, periodEnd, and subject to lists to check if multiple have been supplied.
+        // This is a hack and I hate it. I don't know how to just pull the current url due to
+        // the amount of abstraction going on. I didn't want to waste too much time here.
+        // If there's a better way of doing this, please ping me. - Carter
+        String _periodStart = periodStart.get(0);
+        String _periodEnd = periodEnd.get(0);
+        String _subject = subject.get(0);
+
+        if (periodStart.size() > 1)
+            throw new IllegalArgumentException("Only one periodStart argument can be supplied.");
+
+        if (periodEnd.size() > 1)
+            throw new IllegalArgumentException("Only one periodEnd argument can be supplied.");
+
+        if (subject.size() > 1 || subject.size() <= 0)
+            throw new IllegalArgumentException("You must supply subject argument.");
```

Comment:
Can we move all this into the careGapParamaterValidation method as they are part of validation?